### PR TITLE
[NIC-7] Add Mongock (migrations for MongoDB)

### PR DESCRIPTION
**Pull request by V7074**

### CODE REVIEW

**Review by alelambin — APPROVED** [review]

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/configuration/mongodb/migration/ChangeUnitV1.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.nosql.authservice.entity.SimpleCheckEntity
+import io.mongock.api.annotations.BeforeExecution
+import io.mongock.api.annotations.ChangeUnit
+import io.mongock.api.annotations.Execution
+import io.mongock.api.annotations.RollbackExecution
+import org.slf4j.Logger
+import org.springframework.data.mongodb.core.MongoTemplate
+
+@ChangeUnit(id = CHANGE_UNIT_ID, order = CHANGE_UNIT_ORDER, author = CHANGE_UNIT_AUTHOR)
+class ChangeUnitV1(
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/configuration/mongodb/migration/ChangeUnitV1.kt]: Please rename to entity-oriented style

Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/configuration/mongodb/migration/ChangeUnitV1.kt]: fixed

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt [hunk 2]
```diff
@@ -53,16 +52,15 @@ class DefaultUserComponent(
     }
 
     @Transactional(readOnly = true)
-    override suspend fun getByLoginAndPasswordHash(userDto: UserDto): UserEntity {
-        val (login, passwordHash) = userDto
-        val operationDetails = "Get 'auth' record by login = '$login'"
+    override suspend fun getByLoginAndPasswordHash(user: UserEntity): UserEntity {
+        val operationDetails = "Get 'auth' record by login = '${user.login}'"
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt]: Please add login variable

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt [hunk 3]
```diff
@@ -27,7 +26,7 @@ class DefaultUserComponent(
     private val log: Logger by logger()
 
     @Transactional
-    override suspend fun saveIfNotExists(user: UserEntity): UserEntity {
+    override suspend fun save(user: UserEntity): UserEntity {
         val operationDetails = "Save if not exists 'auth' record with login = ${user.login}"
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt]: Change operation details to "Save 'auth' .."

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt (hunk 2 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt]: fixed

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt (hunk 3 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt]: fixed

**Review by AksyonovaKaterina — APPROVED** [review]

**Pull request merged**